### PR TITLE
Add: File selection acceptance tests

### PR DESCRIPTION
**Pull request by nickanderson**
These tests test that files can be selected by type and by link target.

Ref: https://dev.cfengine.com/issues/6878

### CODE REVIEW

File: tests/acceptance/10_files/10_links/select_file_by_symlink_target.cf [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#######################################################
+#
+# Subdirectories shall not be created when action is
+# warn. Related to bug #926.
```
Comment (nickanderson) [inline tests/acceptance/10_files/10_links/select_file_by_symlink_target.cf]: ack

File: tests/acceptance/10_files/10_links/select_file_by_type.cf [hunk 2]
```diff
@@ -0,0 +1,71 @@
+#######################################################
+#
+# Subdirectories shall not be created when action is
+# warn. Related to bug #926.
```
Comment (nickanderson) [inline tests/acceptance/10_files/10_links/select_file_by_type.cf]: ack

File: tests/acceptance/10_files/10_links/select_file_by_symlink_target.cf [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    OK::
+      "$(this.promise_filename) Pass";
+
+    !OK::
+      "$(this.promise_filename) Fail";
+}
+
+### PROJECT_ID: core
+### CATEGORY_ID: 27
+bundle agent test2
```
Comment (nickanderson) [inline tests/acceptance/10_files/10_links/select_file_by_symlink_target.cf]: ack

File: tests/acceptance/10_files/10_links/select_file_by_symlink_target.cf [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    file_result => "issymlinkto";
+}
+
+body delete nodir
+{
+    rmdirs => "false";
+}
+
+body file_select select_type(type)
+{
```
Comment (nickanderson) [inline tests/acceptance/10_files/10_links/select_file_by_symlink_target.cf]: ack

File: tests/acceptance/10_files/10_links/select_file_by_symlink_target.cf [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+#######################################################
+
+bundle agent test
+{
+  meta:
+    # Windows does not have symlinks, so it does not
+    "test_soft_fail"
+      string => "!windows";
```
Comment (nickanderson) [inline tests/acceptance/10_files/10_links/select_file_by_symlink_target.cf]: ack

**Comment by nickanderson** [comment]
@kacfengine Seems that we fixed the issue between 3.6.3. and 3.6.4. Tests updated per suggestions, now both are passing :)

**Comment by kacfengine** [comment]
Nice, thanks!

**Comment by kacfengine** [comment]
trigger build

**Pull request merged**